src/components/RecipeForm.js

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect } from "react";
+import { IngredientList } from "./Recipe.js";
+import { Title, Container, Button, TextInput } from "./Styles.js";
+import { useHistory, useParams } from "react-router";
+import api from "../data/DataAPI.js";
+
+function RecipeForm(props) {
+  const edit = props.mode === "edit";
+  const history = useHistory();
+  let { recipeId } = useParams();
+
+  let recipe = {
+    name: "",
+    description: "",
+    ingredients: [],
+  };
+
+  const [{ data, error, isLoading }, dispatch] = useState({
+    data: recipe,
+    error: null,
+    isLoading: false,
+  });
+
+  useEffect(() => {
+    //Couldn't find a good way to handle this
+    if (!edit) {
+      return;
+    }
+    dispatch((state) => ({ ...state, isLoading: true }));
+    api
+      .fetchRecipe(recipeId)
+      .then((data) => dispatch({ data, error: null, isLoading: false }))
+      .catch((error) => dispatch({ data: null, error, isLoading: false }));
+  }, [recipeId, edit]);
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    dispatch((prevState) => ({
+      ...prevState,
+      data: {
+        ...prevState.data,
+        [name]: value,
+      },
+    }));
+  };
+
+  const keyPress = (e) => {
+    if (e.keyCode === 13) {
+      const value = e.target.value;
+      dispatch((prevState) => ({
+        ...prevState,
+        data: {
+          ...prevState.data,
+          ingredients: prevState.data.ingredients.concat({ name: value }),
+        },
+      }));
+      e.target.value = "";
+    }
+  };
+
+  const submit = () => {
+    let url = edit
```

Comment:
the same here, I'd move all the logic dealing with the end-points to a separated module